atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/WriteInfo.java

```diff
@@ -49,7 +51,18 @@ default TimestampRangeDelete toDelete(Sweeper sweeper) {
     }
 
     default int toShard(int numShards) {
-        return IntMath.mod(writeRef().cellReference().goodHash(), numShards);
+        return IntMath.mod(dayRotatingHash(), numShards);
+    }
+
+    /**
+     * The purpose of the rotating hash calculation is to redistribute shards every day to alleviate issues caused by
+     * imbalanced write patterns overloading few shards.
+     */
+    default int dayRotatingHash() {
+        int hash = 5381;
+        hash = hash * 31 + writeRef().cellReference().goodHash();
+        hash = hash * 31 + LocalDate.now(ZoneId.of("UTC")).hashCode();
```

Comment:
Discussed offline: this hash isn't good when you have a multiple of 31 shards (which is very common)!